appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy

```diff
@@ -33,22 +33,20 @@ abstract class AbstractAppServiceIntegrationTest extends Specification
 	def addRichard()
 	{
 		def richard = appService.addUser(appService.&assertUserCreationResponseDetails, "R i c h a r d", "Richard", "Quinn", "RQ",
-				"+$timestamp", ["Nexus 6"], [
-					BudgetGoal.createNoGoInstance("news"),
-					BudgetGoal.createNoGoInstance("gambling")
-				])
+				"+$timestamp", ["Nexus 6"])
 		appService.confirmMobileNumber(appService.&assertResponseStatusSuccess, richard)
+		appService.addBudgetGoal(richard, BudgetGoal.createNoGoInstance("news"))
+		appService.addBudgetGoal(richard, BudgetGoal.createNoGoInstance("gambling"))
```

Comment:
There was no other way actually to do this nicely with HAL. At least Spring cannot deserialize a JSON object containing _embedded items I guess :)